### PR TITLE
MAINT: Make selenium optional and use on CircleCI

### PR DESCRIPTION
**Pull request by larsoner**
See title, should make it easier to build our docs. @britta-wstnr can you try building docs on this branch and see if it works? I made the avatar image gen opt-in (and opted CircleCI in) rather than opt-out so you should just be able to check out my branch, and run `make clean` then `make html` and see if it succeeds. I think we should iterate until things work for you to improve our odds of other contributors being able to build docs.

### CODE REVIEW

**Comment by larsoner** [comment]
Will fail until https://github.com/mne-tools/mne-python/pull/12260 lands but local builds now show this instead of the contributor image:

Screenshot from 2023-12-04 13-20-06

File: doc/sphinxext/contrib_avatars.py [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
+        from selenium.webdriver.support.ui import WebDriverWait
+        from selenium.common.exceptions import WebDriverException
+
+        try:
+            options = webdriver.ChromeOptions()
+            options.add_argument("--headless=new")
+            driver = webdriver.Chrome(options=options)
+        except WebDriverException:
+            options = webdriver.FirefoxOptions()
+            options.add_argument("--headless=new")
```
Comment (drammock) [inline doc/sphinxext/contrib_avatars.py:28]: sneaking in a bugfix, `--headless=new` is a chrome option; for firefox it's `-headless` (single dash is correct here)
```suggestion
            options.add_argument("-headless")
```

**Review by drammock — APPROVED** [review]

**Comment by britta-wstnr** [comment]
Thanks @larsoner !
Running `make html-pattern ...` works for me again ! 🎉 

I also tried running a full `make html` but this throws a memory error for me (at `35_dipole_orientations.py`, which ultimately results in `pyvista.plotting.errors.RenderWindowUnavailable: Render window is not current.`and then `sphinx.errors.ExtensionError: Handler <function generate_gallery_rst at 0x16997afc0> for event 'builder-inited' threw an exception (exception: [<mne.viz._brain._brain.Brain object at 0x2a8f0e510>])`).
This surprises me a bit, as I have 16 GB RAM. I will try again tonight after closing everything else down.

**Comment by larsoner** [comment]
Up to date pyvista and pyvistaqt?

I think having the pattern working is enough for this PR at least. But we can continue to discuss here or in a separate issue the full build error. Or just not worry about it since in practice Circle is the only one who needs to suffer through full builds :)

**Comment by britta-wstnr** [comment]
> Up to date pyvista and pyvistaqt?

Yup, fresh conda install (pyvista `0.42.3` and pyvistaqt `0.11.0`)
But indeed, I don't run full builds - so fine for me to not look further into. I just found it puzzling!

**Pull request merged**